pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go

```diff
@@ -236,28 +238,63 @@ func (am *AppManifests) validate() error {
 }
 
 // workload and trait in the same component both have these labels
-func (am *AppManifests) generateCommonLabels(compName, compRevisionName string) map[string]string {
+func (am *AppManifests) generateAndFilterCommonLabels(compName, compRevisionName string) map[string]string {
+	in := func(a string, list []string) bool {
+		for _, b := range list {
+			if b == a {
+				return true
+			}
+		}
+		return false
+	}
+	filter := func(before map[string]string, notAllowedKey []string) map[string]string {
+		after := make(map[string]string)
+		for key, value := range before {
+			if !in(key, notAllowedKey) {
+				after[key] = value
+			}
+		}
+		return after
+	}
```

Comment:
Why not just use a loop inside which invoke `delete(labels, filteredLabelKey)`

Reply:
For each filterLabelKey, it's better to trim spaces before filtering. I think it's highly possible to type spaces between keys in the annotation.

Reply:
yeah, that's better